tests/python_package_test/test_basic.py

```diff
@@ -411,3 +411,18 @@ def test_list_to_1d_numpy(y, dtype):
     result = lgb.basic.list_to_1d_numpy(y, dtype=dtype)
     assert result.size == 10
     assert result.dtype == dtype
+
+
+def test_dump_model():
```

Comment:
Thanks a lot for this enhancement!
Please move this test into `test_engine.py` file because `lgb.train()` function is from `engine.py` module.
Also I think it will be useful to add some asserts into this test. Something like `assert 'leaf_coeff' in dumped_model` for linear model and `assert 'leaf_coeff' not in dumped_model` for ordinary one.